internal/charmstore/store.go

```diff
@@ -1425,6 +1425,36 @@ func (store *Store) Search(sp SearchParams) (SearchResult, error) {
 	return result, nil
 }
 
+// List searches the store for the given ListParams.
+// It returns a SearchResult containing the results of the search.
+func (store *Store) List(sp ListParams) (SearchResult, error) {
+	query := store.DB.Entities().Find(sp.Filters)
+	if sp.sort == nil {
+		query = query.Sort("_id")
+	} else {
+		query = query.Sort(sp.sort...)
+	}
+	var entities []*mongodoc.Entity
+	err := query.All(&entities)
+	if err != nil {
+		return SearchResult{}, errgo.Mask(err)
+	}
+	r := SearchResult{
```

Comment:
I wouldn't be so keen to fit the results into the search result structure. Find something that makes sense for the data you have.